tests/serialize/runstate/dynamodb_state_store_test.py

```diff
@@ -115,221 +57,135 @@ def small_object():
         "run_num": 1,
         "run_time": None,
         "time_zone": None,
         "node_name": "example_node",
-        "runs": [],
         "cleanup_run": None,
         "manual": False,
-        "large_data": [i for i in range(1_000_000)],
+        "runs": [base_run] * 5000,  # gives us a ~2000KB JobRun, ensuring at least a few partitions.
     }
 
 
-@pytest.mark.usefixtures("store", "small_object", "large_object")
+@pytest.mark.usefixtures("store", "job_state_object", "small_jobrun_state_object", "large_jobrun_state_object")
 class TestDynamoDBStateStore:
-    @pytest.mark.parametrize("read_json", [False, True])
-    def test_save(self, store, small_object, large_object, read_json):
-        key_value_pairs = [
-            (
-                store.build_key("job_state", "two"),
-                small_object,
-            ),
-            (
-                store.build_key("job_run_state", "four"),
-                small_object,
-            ),
-        ]
-        store.save(key_value_pairs)
-        store._consume_save_queue()
-
-        assert store.save_errors == 0
-        keys = [
-            store.build_key("job_state", "two"),
-            store.build_key("job_run_state", "four"),
-        ]
-        mock_config = {"read_json.enable": read_json}
-        mock_configuration = staticconf.testing.MockConfiguration(mock_config, namespace="tron")
-        with mock_configuration, mock.patch("tron.config.static_config.load_yaml_file", autospec=True), mock.patch(
-            "tron.config.static_config.build_configuration_watcher", autospec=True
-        ):
-            vals = store.restore(keys)
-        for key, value in key_value_pairs:
-            assert_equal(vals[key], value)
-
-        for key in keys:
-            item = store.table.get_item(Key={"key": key, "index": 0})
-            assert "Item" in item
-            assert "json_val" in item["Item"]
-            assert_equal(json.loads(item["Item"]["json_val"]), small_object)
-
-    def test_delete_if_val_is_none(self, store, small_object, large_object):
-        key_value_pairs = [
-            (
-                store.build_key("job_state", "two"),
-                small_object,
-            ),
-            (
-                store.build_key("job_run_state", "four"),
-                small_object,
-            ),
-        ]
-        store.save(key_value_pairs)
-        store._consume_save_queue()
-
-        delete = [
-            (
-                store.build_key("job_state", "two"),
-                None,
-            ),
-        ]
-        store.save(delete)
-        store._consume_save_queue()
+    @pytest.mark.parametrize(
+        "object_fixture_name",
+        ["small_jobrun_state_object", "large_jobrun_state_object"],
+    )
+    def test_save_and_restore(self, store, object_fixture_name, request):
+        """Verify that objects of different sizes can be saved and restored correctly."""
+        state_object = request.getfixturevalue(object_fixture_name)
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "state_object",
        [small_jobrun_state_object, large_jobrun_state_object],
    )
    def test_save_and_restore(self, store, state_object, request):
        """Verify that objects of different sizes can be saved and restored correctly."""
```

disclaimer: i haven't test that - but shouldn't we be able to use the fixtures directly here rather than using the `getfixturevalue` indirection?